File: learning-service/src/main/java/com/english/learning_service/entity/Plan.java
package com.english.learning_service.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "plan")
public class Plan {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(name = "user_id", nullable = false)
    private String userId;

    private String title;

    private String description;

    @Column(name = "is_completed")
    private boolean isCompleted;

    @Column(name = "start_date")
    private LocalDateTime startDate;

    @Column(name = "end_date")
    private LocalDateTime endDate;

    private Integer target;
}

File: agent-service/app/schemas/plan.py
from typing_extensions import List, TypedDict

class PlanDetail(TypedDict):
    topicType:str
    topicId: str

class PlanGroup(TypedDict):
    name: str
    description: str
    startDate: str
    endDate: str
    details: List[PlanDetail] 
    
class Plan(TypedDict):
    user_info: str
    title: str
    description: str
    startDate: str
    endDate:str
    planGroups: List[PlanGroup]
    
